lib/Models/GNAFApi.js

```diff
@@ -0,0 +1,186 @@
+var loadWithXhr = require('terriajs-cesium/Source/Core/loadWithXhr');
+var defined = require('terriajs-cesium/Source/Core/defined');
+var DeveloperError = require('terriajs-cesium/Source/Core/DeveloperError');
+var defaultValue = require('terriajs-cesium/Source/Core/defaultValue');
+
+var DEFAULT_MAX_RESULTS = 10;
+var MIN_SCORE = 1;
+var DATA61_GNAF_SEARCH_URL = 'http://gnaf.it.csiro.au/es/_search';
+
+/**
+ * Simple JS Api for the Data61 ElasticSearch GNAF service.
+ *
+ * @param corsProxy CorsProxy to use to determine whether calls need to go through the terria proxy.
+ * @param [overrideUrl] The URL to use to query the service - will default if not provided
+ * @param [overriddenLoadWithXhr] loadWithXhr function matching the signature of the one provided by Cesium - provided
```

Comment:
`loadWithXhr` has built in support for unit testing by setting `loadWithXhr.load`.  Also you can use `jasmine.Ajax` to override XMLHttpRequest itself.  So I don't think this extra parameter is necessary.